File: src/scubaLive/SurfaceInterval.java
package scubaLive;

public class SurfaceInterval {
	
	char spg;
	char fpg;
	Dive sdive;
	Dive fdive;
	SurfaceTable stable;
	int interval;

	public SurfaceInterval(Dive dive1, Dive dive2, SurfaceTable table) {
		spg = dive1.getFPG();
		sdive = dive1;
		fdive = dive2;
		interval = 0;
		stable = table;
		spg = dive1.getFPG();
		fpg = spg;
		fdive.updateDive(0, 0, fpg);
		
	}
	
	public void updateInterval(int time) {
		spg = sdive.getFPG();
		interval = time;
		fpg = stable.getEndingPressureGroup(spg, time);
		fdive.updateDive(fpg);
	}
	
	public void updateInterval() {
		this.updateInterval(this.interval);
	}
	
	public void setMinInterval () {
		interval = stable.getSurfaceIntervalTime(sdive.getFPG(), fdive.getMPG())[0];
		updateInterval();
	}

	public char getFPG() {
		return fpg;
	}
	
	public char getSPG() {
		return spg;
	}
	
	public int getInterval() {
		return interval;
	}
	
	@Override
	public String toString() {
		String output = new String();
		output = "Stay on surface for " + interval + " minutes.\n";
		return output;
	}
	
	public void draw() {
		
	}
}

File: client/src/boot/surfaceinterval.js
import SurfaceTable from 'surfacetable.js';
import Dive from 'dive.js';


class SurfaceInterval {

    constructor (dive1, dive2, table) {
        let sdive = dive1;
        let fdive = dive2;
        let interval = 0;
        let stable = table;
        let spg = dive1.getFPG;
        let fpg = spg;
        fdive.updateDive(fpg, 0, 0);
    }

    updateInterval(time) {
        if(time === undefined) time = this.interval;
        this.spg = sdive.getFPG();
        this.interval = time;
        this.fpg = SurfaceTable.getEndingPressureGroup(this.spg, time);
        fdive.updateDive(this.fpg);
    }

    setMinInterval () {
        this.interval = SurfaceTable.getSurfaceIntervalTime(this.sdive.getFPG(), this.fdive.getMPG())[0];
        updateInterval();
    }

    getFPG() {
        return this.fpg;
    }

    getSPG() {
        return this.spg;
    }

    getInterval() {
        return this.interval;
    }


    toString() {
        return "Stay on surface for " + interval + " minutes.\n";
    }

}
